src/retry.ts

```diff
@@ -0,0 +1,41 @@
+import * as core from '@actions/core'
+
+const maxRetries: number = parseInt(core.getInput('retries', { required: false }));
+
+export class Retries {
+	private maxAttemps: number
+
+	constructor() {
+		this.maxAttemps = maxRetries
+	}
+
+	async execute<T>(action: () => Promise<T>): Promise<T> {
+		// run N-1 attempts
+		var attempt = 1;
+		while (attempt < this.maxAttemps) {
+			try {
+				return await action()
+			} catch (err) {
+				core.info( (err as any)?.message )
+			}
+
+			// sleep before retrying
+			const seconds = Math.random() * 10 + 5
```

Comment:
nit: Maybe do exponential backoff here since it's calling into a web service

Reply:
That's a great idea. I'll add that as a second PR to this